Intro-Full/Exercises/kokkoskernels/CGSolve_SpILUKprecond/Begin/cgsolve.cpp

```diff
@@ -155,10 +162,28 @@ int main( int argc, char* argv[] )
     scalar_t zero = scalar_t(0.0);
     scalar_t mone = scalar_t(-1.0);
 
-    // Allocate, read and fill a sparse matrix
-	crsmat_t A        = KokkosKernels::Impl::read_kokkos_crst_matrix<crsmat_t>(afilename.c_str()); //in_matrix
+    // Generate a simple Laplacian matrix on a cartesian grid.
+    // The mat_structure view is used to generate a matrix using
+    // finite difference (FD) or finite element (FE) discretization
+    // on a cartesian grid.
+    // Each row corresponds to an axis (x, y and z)
+    // In each row the first entry is the number of grid point in
+    // that direction, the second and third entries are used to apply
+    // BCs in that direction, BC=0 means Neumann BC is applied,
+    // BC=1 means Dirichlet BC is applied by zeroing out the row and putting
+    // one on the diagonal.
+    Kokkos::View<lno_t*[3], Kokkos::HostSpace> mat_structure("Matrix Structure", 2);
+    mat_structure(0, 0) = nx;  // Request nx grid point in 'x' direction
+    mat_structure(0, 1) = 0;   // Add BC to the left
+    mat_structure(0, 2) = 0;   // Add BC to the right
+    mat_structure(1, 0) = ny;  // Request ny grid point in 'y' direction
+    mat_structure(1, 1) = 0;   // Add BC to the bottom
+    mat_structure(1, 2) = 0;   // Add BC to the top
+
+    crsmat_t A = Test::generate_structured_matrix2D<crsmat_t>("FD", mat_structure);
+
     graph_t  graph    = A.graph; // in_graph
-    const size_type N = graph.numRows();
+    const size_type N = graph.numRows(); std::cout << "Matrix size: " << N << " x " << N << std::endl;
```

Comment:
@vqd8a is the output just for debugging or an output you'd like users to have? Would prefer on a new line.

Reply:
@ndellingwood I'd like users to have. I moved it to a new line.

Reply:
Thanks!